l1-contracts/src/core/libraries/rollup/STFLib.sol

```diff
@@ -168,6 +375,29 @@ library STFLib {
     return getSlotNumber(_blockNumber).epochFromSlot();
   }
 
+  /**
+   * @notice Determines if the chain can be pruned at a given timestamp
+   * @dev Checks whether the proof submission window has expired for the oldest pending blocks.
+   *      Pruning is allowed when:
+   *
+   *      1. There are unproven blocks (pending > proven)
+   *      2. The oldest pending epoch is no longer accepting proofs at the current epoch
+   *
+   *      The proof submission window is defined by the aztecProofSubmissionEpochs configuration,
+   *      which specifies how many epochs after an epoch ends that proofs are still accepted.
+   *
+   *      Example timeline:
+   *      - Block proposed in epoch N
+   *      - Proof submission window = 2 epochs
+   *      - Proof deadline = end of epoch N+2
+   *      - If current time > epoch N+2, pruning is allowed
```

Comment:
As above, not current time, but provided `_ts`. Also think this is 1-off.

If the value `proof submission window = 1 epoch` it would be prunable when `epoch(ts) >= (N+1+1)`, The proof submission window is essentially "how many epochs *after* N do you have".